### DISCUSSION TITLE
el código me arroja el resultado, pero no me da un 55

### DISCUSSION THREAD

**Discussion by JoHpacheco** [rramosp/ai4eng.v1#308] [04 - GESTIÓN DE DATOS]
además de eso, pongo los valores sin decimales y lo mismo, he usado otros códigos y nada
import pandas as pd
train_df = pd.read_csv('train.csv')
building_metadata_df = pd.read_csv('building_metadata.csv')
weather_train_df = pd.read_csv('weather_train.csv')
train_df

building_train_info_df = train_df[(train_df['meter'] == 0) & (train_df['building_id'] == bid) & (train_df['timestamp'].str.contains(date))]
building_metadata_info_df = building_metadata_df[building_metadata_df['building_id'] == bid]

site_id = building_metadata_info_df['site_id'].values[0]
building_weather_info_df = weather_train_df[(weather_train_df['site_id'] == site_id) & (weather_train_df['timestamp'].str.contains(date))]

building_metadata_dict = building_metadata_info_df.to_dict('records')[0]
building_train_metadata_df = pd.concat([
        building_train_info_df,
        building_metadata_info_df
    ],axis=0, ignore_index=True).fillna(value=building_metadata_dict)[:-1]

building_info_df = pd.concat([
        building_train_metadata_df,
        building_weather_info_df.reset_index()
    ], axis=1)
metadata_int_columns = ['building_id', 'meter', 'meter_reading', 'site_id', 'square_feet']
for col in metadata_int_columns:
  building_info_df[col] = building_info_df[col].astype(int)
building_info_df = building_info_df.loc[:,~building_info_df.columns.duplicated()]
building_info_df

selected_columns = ['meter_reading', 'site_id', 'air_temperature', 'cloud_coverage',
'dew_temperature', 'precip_depth_1_hr', 'sea_level_pressure',
'wind_direction', 'wind_speed', 'square_feet', 'year_built']
selected_info_df = building_info_df[selected_columns].fillna(0)
meter_reading_sum = building_info_df['meter_reading'].sum()
important_variables_sum = selected_info_df.iloc[:, 1:].sum().sum()
important_variables_sum, meter_reading_sum
selected_info_df

# target variable sum
sumY = meter_reading_sum

# input variables sum
sumX = important_variables_sum

sumX, sumY
(297698.3, 180)

task_01 submitted. your grade is 0
----- grader message -------
checking for building_id=938 and date='2016-02-19'
incorrect

**Comment by Felipe-RA** [comment]
Saludos @JoHpacheco 

En el código que comentas, tu resultado tiene decimales. Si tienes otra versión en la cuál este no es el caso es más apropiado que compartas esa.

Por otro lado, para verificar si tu código corre bien, te sugiero que pruebes cambiando al `bid`  y `date` de prueba para verificar que tu flujo de trabajo de el resultado correcto:

Recuerda que los valores de prueba son:

```python
bid = 900
date = "2016-02-01"
```

y te debería dar para esos valores de prueba:

```python
sumX = 2705422
sumY = 4941
```

**Comment by JoHpacheco** [comment]
hola profe como estas, la verdad ningun codigo me da 5 y siempre me da lo
mismo he probado con códigos de comapeños y anda le comparto
https://colab.research.google.com/drive/10WjNHQofIJmOnunkdcWO_33MPiATPkDv?usp=sharing
cordialmente

Joseph Pacheco

Estudiante  de Ingeniería Ambiental


El vie, 5 abr 2024 a las 21:30, Felipe R.A ***@***.***>)
escribió:

> Saludos @JoHpacheco <https://github.com/JoHpacheco>
>
> En el código que comentas, tu resultado tiene decimales. Si tienes otra
> versión en la cuál este no es el caso es más apropiado que compartas esa.
>
> Por otro lado, para verificar si tu código corre bien, te sugiero que
> pruebes cambiando al bid y date de prueba para verificar que tu flujo de
> trabajo de el resultado correcto:
>
> Recuerda que los valores de prueba son:
>
> bid = 900date = "2016-02-01"
>
> y te debería dar para esos valores de prueba:
>
> sumX = 2705422sumY = 4941
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/308#discussioncomment-9027283>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BGPJL4KH7BXAEKMLZHNPWYTY35M3LAVCNFSM6AAAAABFZ6SF5SVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TAMRXGI4DG>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 


"La información aquí contenida es para uso exclusivo de la persona o 
entidad de destino. Está estrictamente prohibida su utilización, copia, 
descarga, distribución, modificación y/o reproducción total o parcial, sin 
el permiso expreso de Universidad de Antioquia, pues su contenido puede ser 
de carácter confidencial y/o contener material privilegiado. Si usted 
recibió esta información por error, por favor contacte en forma inmediata a 
quien la envió y borre este material de su computador. Universidad de 
Antioquia no es responsable por la información contenida en esta 
comunicación, el directo responsable es quien la firma o el autor de la 
misma."


-- 
UdeA

**Comment by Felipe-RA** [comment]
@JoHpacheco 

No me refería a que te diese 5, me refiero a la versión de código que me indicas usas para poner los valores sin decimales:

> además de eso, pongo los valores sin decimales y lo mismo, he usado otros códigos y nada

Porque en tu notebook veo que tienes como resultado un número flotante para `sumX`

```python
sumX, sumY

(297698.3, 180)
```

Nuevamente, te  invito a que pruebes con el código que me dices te da un entero y verifiques si los valores de prueba te dan correctos:

Por otro lado, para verificar si tu código corre bien, te sugiero que pruebes cambiando al `bid`  y `date` de prueba para verificar que tu flujo de trabajo de el resultado correcto:

Recuerda que los valores de prueba son:

>```python
>bid = 900
>date = "2016-02-01"
>```
>
>y te debería dar para esos valores de prueba:
>
>```python
>sumX = 2705422
>sumY = 4941
>```
